File: 0118-pascals-triangle/0118-pascals-triangle.cs
public class Solution {
    public IList<IList<int>> Generate(int numRows) {
        var result = new List<IList<int>>();
        var row = new List<int>();                  // this same variable will be reused in all layers
        for(int j=0; j< numRows; j++){              // for all the layers
            for(int i=row.Count -1; i>0; i--){      // starting from right side
                row[i] = row[i] + row[i-1];         // the current row already has prev row data
            }
            row.Add(1);                             // add 1 at end of each row
            result.Add(new List<int>(row));
        }
        return result;
    }
}

File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> result = new ArrayList<>();
        List<Integer> row = new ArrayList<>();                // this same variable will be reused in all layers

        for (int j = 0; j < numRows; j++) {                   // for all the layers
            for (int i = row.size() - 1; i > 0; i--) {        // starting from right side
                row.set(i, row.get(i) + row.get(i - 1));      // the current row already has prev row data
            }
            row.add(1);                                       // add 1 at end of each row
            result.add(new ArrayList<>(row));
        }
        return result;
    }
}
